instrument/extended.go

```diff
@@ -21,26 +21,99 @@
 package instrument
 
 import (
+	"fmt"
 	"os"
 	"runtime"
+	"strings"
 	"sync/atomic"
 	"time"
 
-	"github.com/shirou/gopsutil/process"
+	"github.com/m3db/m3x/process"
+
 	"github.com/uber-go/tally"
 )
 
+// ExtendedMetricsType is a type of extended metrics to report.
+type ExtendedMetricsType int
+
+const (
```

Comment:
Talked offline, the simple metrics (goroutine count etc) are generally nice to have and cheap to report so leaving it as is for now.